Add options needed to build plugins

gdesmott opened the pull request:
See #135 for the context and API discussion.

lu-zero reviewed (approved):
It seems good to me, I'm a little busy today so I'd merge it once I give it a go on sled, rav1e and nnnoiseless. If you already tested it it can be merged now.

sdroege commented:
Also ebur128 please, it makes use of many of the other configurations that are hopefully not affected by this change :)

gdesmott commented:
I never build or use any of those crates so I prefer to let you double check I didn't break anything. :)

gdesmott commented:
Just found out that `Libs` field in the pkg-config file should be updated with the subdirectory as well. Let me fix that before merging. :)

gdesmott commented:
> Just found out that `Libs` field in the pkg-config file should be updated with the subdirectory as well. Let me fix that before merging. :)

Fixed.

I also stopped settings `Cflags` when we are not generating any headers and added a extra commit preventing the creation of `bin` directory if not needed.

lu-zero commented on src/pkg_config_gen.rs:
```diff
@@ -51,6 +51,23 @@ impl PkgConfig {
         };
 
+        let libdir = if let Some(subdir) = &capi_config.library.install_subdir {
+            format!("{}{}{}", "${libdir}", MAIN_SEPARATOR, subdir)
```
probably we could avoid using `format!` and just append to the path and convert later.

gdesmott replied:
Indeed that's easier, done.
